src/Typesense/TypesenseClient.cs

```diff
@@ -41,13 +43,26 @@ public TypesenseClient(IOptions<Config> config, HttpClient httpClient)
         httpClient.BaseAddress = new Uri($"{node.Protocol}://{node.Host}:{node.Port}");
         httpClient.DefaultRequestHeaders.Add("X-TYPESENSE-API-KEY", config.Value.ApiKey);
         _httpClient = httpClient;
+        if (customJsonSerializerOptions is not null)
+        {
+            _jsonNameCaseInsensitiveTrue = new JsonSerializerOptions(customJsonSerializerOptions)
+            {
+                PropertyNameCaseInsensitive = true
+            };
+
+            _jsonOptionsCamelCaseIgnoreWritingNull = new JsonSerializerOptions(customJsonSerializerOptions)
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull
+            };
```

Comment:
Why is this different than when `customJsonSerializationOptions` are null?

So if custom serialization options are added, CaleCase and JsonIgnoreWhenWritingNull is set, why is that?